File: Matlab_Octave/example3_bouncingrod/dynamics/anchor.m
function [v, isterm, dir, da] = anchor(x, params)
% A Computes the anchor for the bouncing rod
%
% [V, ISTERM, DIR, DA] = ANCHOR(X,PARAMS) returns V, ISTERM, and DIR 
% according to the ode45 event function documentation. V is zero at apex.

% Event is detected if apex
v = x(3);

isterm = 1;
dir    = -1;

da     = [0,0,1,0];
end

File: Julia/example3_bouncingrod/dynamics/anchor.jl
function anchor(x, t, integrator, params)
    # A Computes the anchor for the bouncing rod
    #
    # V = ANCHOR(X,PARAMS) returns V, ISTERM, and DIR 
    # according to the ode45 event function documentation. V is zero at apex.

    # Event is detected if apex
    v = x[3]

    return v
end